src/uvw/emitter.hpp

```diff
@@ -193,7 +194,7 @@ class Emitter {
 
         return (!(type < handlers.size()) ||
                 !handlers[type] ||
-                static_cast<Handler<E>&>(*handlers[type]).empty());
+                handlers[type]->empty());
```

Comment:
This is not an improvement.
We know the exact type of the handler (that is `E`) and casting it allows a direct call to the method in the final class.
On the other side, you are proposing to call a virtual method, thus to introduce a double dispatch that gives no benefits.
I'd suggest to rollback this change.